### PR TITLE
[DOCFIX] Update YARN deploy doc

### PR DESCRIPTION
**Pull request by aaudiber**

### CODE REVIEW

File: docs/en/deploy/Running-Alluxio-On-Yarn.md [hunk 1]
```diff
@@ -57,10 +59,6 @@ $ ${HADOOP_HOME}/sbin/start-yarn.sh
 
 ## Alluxio YARN integration
```
Comment (apc999) [inline docs/en/deploy/Running-Alluxio-On-Yarn.md]: maybe calling this "Running Alluxio as YARN application"? `Alluxio YARN integration` is not very informative in doc like this.

File: docs/en/deploy/Running-Alluxio-On-Yarn.md [hunk 2]
```diff
@@ -6,14 +6,16 @@ group: Deploying Alluxio
...
-- How to find logs when Alluxio containers fail to start
-- How to diagnose when there aren't enough YARN resources to launch Alluxio containers
+This document discusses how to run Alluxio in a YARN environment. There are two
+approaches: running Alluxio standalone, and running Alluxio as an application
+on top of YARN. We support both approaches, but recommend running Alluxio
+standalone for a few reasons:
+
+- Alluxio's architecture is designed to have one worker per host, but this is not
+supported by YARN.
+- Running Alluxio on top of YARN adds the additional complexity of managing Alluxio's
```
Comment (apc999) [inline docs/en/deploy/Running-Alluxio-On-Yarn.md]: how about fault tolerance? I remember we don't support restart workers with Alluxio on YARN

File: docs/en/deploy/Running-Alluxio-On-Yarn.md [hunk 3]
```diff
@@ -6,14 +6,16 @@ group: Deploying Alluxio
...
-TODO: Add troubleshooting help
-- How to debug when alluxio-yarn.sh fails to bring up Alluxio servers
-- How to find logs when Alluxio containers fail to start
-- How to diagnose when there aren't enough YARN resources to launch Alluxio containers
+This document discusses how to run Alluxio in a YARN environment. There are two
+approaches: running Alluxio standalone, and running Alluxio as an application
+on top of YARN. We support both approaches, but recommend running Alluxio
+standalone for a few reasons:
+
+- Alluxio's architecture is designed to have one worker per host, but this is not
```
Comment (apc999) [inline docs/en/deploy/Running-Alluxio-On-Yarn.md]: mentioning something like: fundamentally, Alluxio is a long-running daemon like HDFS. HDFS is not supposed to be run on YARN

File: docs/en/deploy/Running-Alluxio-On-Yarn.md [hunk 4]
```diff
@@ -6,14 +6,16 @@ group: Deploying Alluxio
...
-TODO: Expand the explanation of why we prefer to run alongside YARN instead of on top. Relevant
-mailing list thread: https://groups.google.com/forum/#!topic/alluxio-users/KEvBE9AiJlE
-TODO: Tell users how to configure Alluxio servers running on Yarn
-TODO: Discuss how to configure the YARN application classpath so that applications can work with Alluxio.
-TODO: Add troubleshooting help
-- How to debug when alluxio-yarn.sh fails to bring up Alluxio servers
-- How to find logs when Alluxio containers fail to start
-- How to diagnose when there aren't enough YARN resources to launch Alluxio containers
+This document discusses how to run Alluxio in a YARN environment. There are two
+approaches: running Alluxio standalone, and running Alluxio as an application
```
[the remark this answers is not in the document]
Comment (aaudiber) [reply docs/en/deploy/Running-Alluxio-On-Yarn.md]: good suggestions, done

File: docs/en/deploy/Running-Alluxio-On-Yarn.md [hunk 5]
```diff
@@ -6,14 +6,16 @@ group: Deploying Alluxio
...
+on top of YARN. We support both approaches, but recommend running Alluxio
```
[the remark this answers is not in the document]
Comment (aaudiber) [reply docs/en/deploy/Running-Alluxio-On-Yarn.md]: added to the introduction in the running "alluxio as an application on yarn" section

File: docs/en/deploy/Running-Alluxio-On-Yarn.md (hunk 3 above)
Comment (aaudiber) [reply docs/en/deploy/Running-Alluxio-On-Yarn.md]: added

File: docs/en/deploy/Running-Alluxio-On-Yarn.md (hunk 2 above)
Comment (aaudiber) [reply docs/en/deploy/Running-Alluxio-On-Yarn.md]: added

File: docs/en/deploy/Running-Alluxio-On-Yarn.md (hunk 1 above)
Comment (aaudiber) [reply docs/en/deploy/Running-Alluxio-On-Yarn.md]: updated

File: docs/en/deploy/Running-Alluxio-On-Yarn.md [hunk 6]
```diff
@@ -70,14 +68,7 @@ cluster.
...
 
-## Configuration
-
-To customize Alluxio master and worker with specific properties (e.g., tiered storage setup on each
-worker), see [Configuration settings](Configuration-Settings.html). To ensure your configuration can be
-read by both the ApplicationMaster and Alluxio master/workers, put `alluxio-site.properties` in
-`/etc/alluxio/alluxio-site.properties`.
-
-## Run Alluxio Application
+## Launch Alluxio
```
[the remark this answers is not in the document]
Comment (aaudiber) [reply docs/en/deploy/Running-Alluxio-On-Yarn.md]: fixed, thanks

**Comment by aaudiber** [comment]
@apc999 Thanks for the review. I've addressed your comments, PTAL

File: docs/en/deploy/Running-Alluxio-On-Yarn.md [hunk 7]
```diff
@@ -6,16 +6,20 @@ group: Deploying Alluxio
...
-## Standalone
+- Alluxio servers are long-running daemons like HDFS. HDFS is not supposed to run on YARN.
+- Alluxio's architecture is designed to have one worker per host, but this is not
+well supported by YARN.
+- Running Alluxio on top of YARN adds the additional complexity of managing Alluxio's
+ApplicationMaster.
+- Multi-master high availability mode is not currently supported by our YARN integration.
+- The YARN integration does not support restarting workers in case they crash.
+
+## Running Alluxio Alongside YARN
```
Comment (apc999) [inline docs/en/deploy/Running-Alluxio-On-Yarn.md]: To run Alluxio alongside YARN, we need to ensure YARN apps can talk to Alluxio. What's the requirement or common practice? e.g., do we need to set the "yarn.application.classpath" property to include Alluxio client jar?

Comment (aaudiber) [reply docs/en/deploy/Running-Alluxio-On-Yarn.md]: added

**Pull request merged**